device_doctor/test/src/health_test.dart

```diff
@@ -0,0 +1,172 @@
+      }
+      HealthCheckResult result = await removeCachedData(pf: pf, fs: fs);
+      for (String dir in folders) {
+        expect(await fs.directory(dir).exists(), isFalse);
+      }
+      expect(result.succeeded, true);
+    });
+
+    test('removes cache directories not all exist', () async {
+      platform.FakePlatform pf = platform.FakePlatform()
+        ..operatingSystem = "macos"
+        ..environment = <String, String>{"HOME": "/foo"};
+      String dir = '/foo/.dartServer';
+      fs.directory(dir)..createSync(recursive: true);
+      HealthCheckResult result = await removeCachedData(pf: pf, fs: fs);
+      expect(await fs.directory(dir).exists(), isFalse);
+      expect(result.succeeded, true);
+    });
+  });
+
+  group('testCloseIosDialog', () {
+    MockProcessManager pm;
+    DeviceDiscovery discovery;
+
+    setUp(() async {
+      pm = MockProcessManager();
+      discovery = FakeIosDeviceDiscovery();
+    });
+
+    test('succeeded', () async {
+      Process proc = FakeProcess(0);
+      when(pm.start(any, workingDirectory: anyNamed('workingDirectory'))).thenAnswer((_) => Future.value(proc));
+
+      HealthCheckResult res = await closeIosDialog(pm: pm, discovery: discovery);
+
+      expect(res.succeeded, isTrue);
+    });
+
+    test('succeeded with code signing overwrite', () async {
+      Process proc = FakeProcess(0);
+      when(pm.start(any, workingDirectory: anyNamed('workingDirectory'))).thenAnswer((_) => Future.value(proc));
+      platform.Platform pl = platform.FakePlatform(environment: <String, String>{
+        'FLUTTER_XCODE_CODE_SIGN_STYLE': 'Manual',
+        'FLUTTER_XCODE_DEVELOPMENT_TEAM': 'S8QB4VV633',
+        'FLUTTER_XCODE_PROVISIONING_PROFILE_SPECIFIER': 'a name with space',
+      });
+
+      HealthCheckResult res = await closeIosDialog(pm: pm, discovery: discovery, pl: pl);
+
+      expect(res.succeeded, isTrue);
+    });
+
+    test('failed', () async {
+      Process proc = FakeProcess(123);
+      when(pm.start(any, workingDirectory: anyNamed('workingDirectory'))).thenAnswer((_) => Future.value(proc));
+
+      expect(
+        closeIosDialog(pm: pm, discovery: discovery),
+        throwsA(TypeMatcher<BuildFailedError>()),
+      );
+    });
+  });
+}
+
+class FakeIosDeviceDiscovery extends Fake implements DeviceDiscovery {
+  @override
+  Future<List<Device>> discoverDevices({int retriesDelayMs = 10000}) {
+    IosDevice d = IosDevice(deviceId: 'fakeDeviceId');
+    return Future.value(<Device>[d]);
+  }
+}
+
+class MockProcessManager extends Mock implements ProcessManager {}
+
+class FakeProcess extends Fake implements Process {
+  FakeProcess(int exitCode) : _exitCode = exitCode;
+
+  int _exitCode;
+
+  @override
+  Future<int> get exitCode => Future.value(_exitCode);
+
+  @override
+  Stream<List<int>> get stderr => Stream.fromIterable([
+        [1, 2, 3]
+      ]);
+
+  @override
+  Stream<List<int>> get stdout => Stream.fromIterable([
+        [1, 2, 3]
+      ]);
+}
```

Comment:
Nice work with the tests! It's great to see your improvements to testing infra tools, and it will pay off with less infra outages.